controllers/provider-gcp/pkg/internal/infrastructure/terraform.go

```diff
@@ -61,25 +65,38 @@ func ComputeTerraformerChartValues(
 	cluster *extensionscontroller.Cluster,
 ) map[string]interface{} {
 	var (
-		vpcName   = DefaultVPCName
-		createVPC = true
+		vpcName           = DefaultVPCName
+		createVPC         = true
+		createCloudRouter = true
+		cloudRouterName   string
 	)
 
 	if config.Networks.VPC != nil {
 		createVPC = false
+		createCloudRouter = false
+
 		vpcName = config.Networks.VPC.Name
+
+		cloudRouterName = "none"
+		if config.Networks.VPC.CloudRouter != nil && len(config.Networks.VPC.CloudRouter.Name) > 0 {
+			cloudRouterName = config.Networks.VPC.CloudRouter.Name
+		}
 	}
 
 	return map[string]interface{}{
 		"google": map[string]interface{}{
 			"region":  infra.Spec.Region,
 			"project": account.ProjectID,
 		},
-		"create": map[string]interface{}{
-			"vpc": createVPC,
-		},
 		"vpc": map[string]interface{}{
 			"name": vpcName,
+			"cloudRouter": map[string]interface{}{
```

Comment:
only provide this key to the terraform helm chart if a cloud router name was given

Reply:
as discussed, none would be better discriptive to write in the terraform output, so it will be set to none if not given.

Reply:
I don't agree. What if a user has named its cloud router "none"? It's always better to set the pointer object to `nil` if the value is optional, see https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required.

Reply:
I am aware of the conventions in this case :). But this was decided because of the terrafom output parsing when we discussed earlier. In anycase, if you still think having an empty string in the terraform output is okay then we can replace none.

Reply:
Why do you want to have an output value at all in case no cloud router was created? I would simply omit the output value (<=> no router was created).